File: MSFE_SE184931/src/types/blindBox.ts
import type { BlindBoxCategory } from "./blindBoxCategory";

export interface BlindBox {
  id: number;
  name: string;
  category: BlindBoxCategory;
  brandId: number;
  rarity: string;
  price: number;
  releaseDate: string;
  stock: number;
}

export interface BlindBoxRequest {
  name: string;
  categoryId: number;
  brandId: number;
  rarity: string;
  price: number;
  stock: number;
}

File: MSBrand_SE184931/src/main/java/org/example/msbrand_se184931/entities/BlindBox.java
package org.example.msbrand_se184931.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Table(name = "blind_boxes")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class BlindBox {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String name;

    private int categoryId;

    @ManyToOne
    @JoinColumn(name = "brand_id")
    private Brand brand;

    private String rarity;

    private double price;

    private LocalDate releaseDate;

    private int stock;
}

